### ISSUE TITLE
Endcap asymmetry

### ISSUE THREAD

**Issue by ekneale** [WCSim/WCSim#484]
There is something causing asymmetry in the endcaps. When generating photons with WCSimLIGen, I found asymmetry when plotting charge maps (diffuser open angle 40 degrees):

![Screenshot from 2024-10-18 17-02-24](https://github.com/user-attachments/assets/a00f3795-2096-42c2-9c11-9feec0bd453a)
Top endcap, central diffuser at (0,0,+3305.82)

![Screenshot from 2024-10-18 17-08-26](https://github.com/user-attachments/assets/5c7856f0-d725-4706-984d-66acb75443d7)
Bottom endcap, central diffuser at (0,0,-3305.82)

![Screenshot from 2024-10-17 10-12-20](https://github.com/user-attachments/assets/d5293741-f486-4a74-aa71-a62490a29e4e)
Bottom endcap, centre edge diffuser

I have checked amongst other things:

1. Using hard-coded diffuser angles in WCSimLIGen in case there was an error reading in the angles.
2. Generated photon angles - these are in line with the diffuser angles e.g. injector angles of theta = 0, phi = 0, with a direction (0,0,-1) sets all photon directions to (0,0,-1) as expected. 
3. Deactivating WLS plates

Photon visualisation looks fine:

![Screenshot from 2024-10-17 12-59-29](https://github.com/user-attachments/assets/51cb9229-9cbf-44e2-89fe-d198c87fa9c2)

If I set the injector open angle to zero and point it straight down, I get the same asymmetry. However if I use the old geometry it looks pretty symmetrical:

![Screenshot from 2024-10-22 17-15-43](https://github.com/user-attachments/assets/8f5417c7-e8b1-45e7-a6ce-0cb37789bbc8)

I can't find any obvious overlaps in the new geometry but will continue to investigate.

**Comment by tdealtry** [comment]
Thanks for opening the issue @ekneale. I know we've spoken a bit offline, but a few more possibilities to check

- is the same G4-defined material used for top & bottom? In terms of tyvek, the way the code is setup (applying it to the surface of a cylinder) this is unlikely, but @patrickstowell should confirm that my understanding is correct
- are all rotations applied correctly? e.g. are the top PMTs/WLS plates "upside down"? I think the easiest way to check is to visual it (@patrickstowell @kmtsui know how to visualise geometry in WCSim. I've never got it working personally)
- is there some kind of offset of any aspect of the geometry away from being symmetrical about `z=0`? Not sure if visualisation has some nice co-ordinate features to help check this?
- it'd be useful to confirm which version you're using (e.g. commit hash or tag)
- what geometry options are you using? From what you said privately, I was unclear which geometry options you were using. I think it'd be useful to have output from (at least) these 3 cases, to check if it's an OD option that's causing issues (Warning: a lot of features in `geom_hk_od_options.mac` are untested with the realisitic geometry. Some shouldn't do anything, but I can't guarantee anything unexpected (e.g. PMTs per cell)) (Do also note the position of `/WCSim/Construct`, and that the OD options come after the `/WCSim/WCgeom` line)
  - Vanilla "realistic"
    ```
    /WCSim/WCgeom HyperK_HybridmPMT_WithOD_Realistic
    # no other geometry options
    /WCSim/Construct
    ```
  - "Realistic" + 3inch PMT
    ```
    /WCSim/WCgeom HyperK_HybridmPMT_WithOD_Realistic
    /WCSim/HyperKOD/PMTODRadius 3inch
    /WCSim/Construct
    ```
  - "Realistic" + full OD options mac file.
    ```
    /WCSim/WCgeom HyperK_HybridmPMT_WithOD_Realistic
    /control/execute macros/geom_hk_od_options.mac
    /WCSim/Construct
    ```
- i've sent you some info regarding a "particle gun imaging" script setup I used previous

**Comment by kmtsui** [comment]
For detector visualization, I am usually using the `OGLSX` option, like `macros/visOGLSX.mac`. This requires `GEANT4_USE_OPENGL_X11=ON` and relevant libraries be available in geant4 installation.

For the histograms, can you check whether each bin contains the same number of PMTs? It could be a plotting artifact if some bin contains more PMTs than others.

**Comment by ekneale** [comment]
Thanks @kmtsui, I'll try to get the visualisation working with the PMTs and WLS plates. For the histograms, each bin represents a single PMT so it's not an artefact of binning. I've looked quite carefully for bugs in the plotting code but there's nothing obvious and since I get a symmetrical distribution in the plots of sims using the old geometry I think I can rule out an issue with the plotting but I'll keep checking.

**Comment by ekneale** [comment]
Resolved by PR #486 :

![image](https://github.com/user-attachments/assets/374a33f7-6939-48a7-ac35-4a47e0e74785)
Bottom endcap diffuser
